uploader/google_music_manager_uploader/uploader_daemon.py

```diff
@@ -51,15 +46,22 @@ def upload(directory='.', oauth=os.environ['HOME'] + '/oauth', remove=False, upl
             uploaded, matched, not_uploaded = api.upload(file_path, True)
             if remove and (uploaded or matched):
                 os.remove(file_path)
-    observer = Observer()
-    observer.schedule(event_handler, directory, recursive=True)
-    observer.start()
-    try:
-        while True:
-            time.sleep(1)
-    except KeyboardInterrupt:
-        observer.stop()
-    observer.join()
+
+    if not oneshot:
```

Comment:
This is generally a bad idea to have such cyclomatic complexity. Here's a better way to implement the same thing

```python
if oneshot:
    sys.exit(0)
```